### ISSUE TITLE
Inspection Report - Capability to Edit the IR title

### ISSUE THREAD

**Issue by franck-boullier** [unee-t/frontend#444]
# The problem:

It is not possible to edit the title of an Inspection Report

# The Solution:

Allow users who are allowed to access an IR to change the IR title while the IR is still in draft mode
The design for this exists already in the RTB

![image](https://user-images.githubusercontent.com/31331637/44501741-11109b00-a6c1-11e8-9862-12c1f7c33e6b.png)

**Comment by nbiton** [comment]
Why is there a button to make this field editable, but there's no button to confirm?

If this means the field's value is constantly saved, why not make always editable in the fir

**Comment by franck-boullier** [comment]
> Why is there a button to make this field editable, but there's no button to confirm?

@nbiton you're right: there should be a button to confirm/save after the title is edited.

@kiatlim and @michkwt can you give us a suggestion for the icon/UI to use there?

**Comment by franck-boullier** [comment]
> If this means the field's value is constantly saved, why not make always editable in the fir

This is not a desirable solution ---> 
- the user will not be sure if/when the title is updated
- This might send several notification of case update and this is an issue (similar problem when a user changes the `solution` or the `next step` today...

**Comment by kiatlim** [comment]
@nbiton for the edit, do we have something similar on the edit case fields? if there is, I would prefer it to be edit inline than edit in the dialog. 

When editing inline, we can use a check mark as an indicator of save / done / complete editing.

What do you think?

**Comment by nbiton** [comment]
@kiatlim Currently the case's title is placed in an always editable field, every single keystroke triggers saving of the title (although we throttle fast keystrokes so it is saved only once if typed fast).
It doesn't seem like the ideal solution either, due to the reasons @franck-boullier mentioned, so I suppose we'd like to move away from that editing paradigm (it is also used for case's "solution and "next steps"). If we'd still like to do editing inline, we might want to adopt a similar experience to the way GH issue title are edited.

**Comment by franck-boullier** [comment]
> If we'd still like to do editing inline, we might want to adopt a similar experience to the way GH issue title are edited.

That sounds like a good approach to me!

**Comment by kiatlim** [comment]
@nbiton for material, it seems like the "saving" is done when the user defocus out from the field.
https://material-ui.com/demos/text-fields/

But in any case, the github approach is cool for me too, we just need to display the save / cancel (i suggest using tick / cross.

**Comment by franck-boullier** [comment]
> I suggest using tick / cross.

This has my vote too!

**Comment by nbiton** [comment]
@kiatlim I'm not sure which one of the examples demonstrates that. But it won't be a good approach for us anyway, as the user might close the tab/ exit the browser immediately after editing, so it might not be saved.

Would it be possible to make a quick mock up to how the second approach would look like?

**Comment by michkwt** [comment]
<img width="363" alt="screen shot 2018-09-04 at 12 17 59 pm" src="https://user-images.githubusercontent.com/38183796/45009811-c8e86580-b03c-11e8-8a95-33f509c93880.png">
<img width="353" alt="screen shot 2018-09-04 at 12 18 04 pm" src="https://user-images.githubusercontent.com/38183796/45009808-c84fcf00-b03c-11e8-90f4-250c435f9640.png">
<img width="352" alt="screen shot 2018-09-04 at 12 18 10 pm" src="https://user-images.githubusercontent.com/38183796/45009809-c8e86580-b03c-11e8-937e-91a2ee27bee3.png">

We explored several options above and would recommend Option 1 which seem to have clear and straightforward CTA. 
Note: when in edit mode for the title, the save Inspection Report icon will not be there. 

@franck-boullier @nbiton

**Comment by franck-boullier** [comment]
I vote for option 1 too: this is the clearest call to action IMO

**Comment by nbiton** [comment]
I like option 1 too
